packages/plugin-essentials/sources/commands/unlink.ts

```diff
@@ -0,0 +1,104 @@
+import {BaseCommand, WorkspaceRequiredError}                                 from '@yarnpkg/cli';
+import {Cache, Configuration, miscUtils, Project, StreamReport, structUtils} from '@yarnpkg/core';
+import {npath, ppath}                                                        from '@yarnpkg/fslib';
+import {Command, Option, Usage, UsageError}                                  from 'clipanion';
+import micromatch                                                            from 'micromatch';
+
+// eslint-disable-next-line arca/no-default-export
+export default class UnlinkCommand extends BaseCommand {
+  static paths = [
+    [`unlink`],
+  ];
+
+  static usage: Usage = Command.Usage({
+    description: `disconnect the local project from another one`,
+    details: `
+      This command will remove any resolutions in the project-level manifest that would have been added via a yarn link with similar arguments.
+    `,
+    examples: [[
+      `Unregister a remote workspace in the current project`,
+      `$0 unlink ~/ts-loader`,
+    ], [
+      `Unregister all workspaces from a remote project in the current project`,
+      `$0 unlink ~/jest --all`,
+    ], [
+      `Unregister all previously linked workspaces`,
+      `$0 unlink --all`,
+    ], [
+      `Unregister all workspaces matching a glob`,
+      `$0 unlink '@babel/*'`,
+    ]],
+  });
+
+  all = Option.Boolean(`-A,--all`, false, {
+    description: `Unlink all workspaces belonging to the target project from the current one`,
+  });
+
+  leadingArgument = Option.String({
+    required: false,
+  });
```

Comment:
Not a blocker, but I'd expect this command to instead take a Rest parameter so that users can pass multiple patterns (similar to how most other commands work, for instance `yarn remove`).

Reply:
https://github.com/yarnpkg/berry/pull/2931